python/ray/_private/utils.py

```diff
@@ -521,6 +531,7 @@ def get_num_cpus() -> int:
             if (
                 "RAY_DISABLE_DOCKER_CPU_WARNING" not in os.environ
                 and "KUBERNETES_SERVICE_HOST" not in os.environ
+                and not override_docker_warning
```

Comment:
Can you add a CLI/output test for this?

Reply:
sgtm
Will add a test that that calls the function with this input, mocks the log method, and asserts not called.

Reply:
sounds good, btw as a nit, maybe the default for the arg should just be `os.environ.get(...)` for clarity?

Reply:
Great idea!

Reply:
Added default arg.
Added full tautological glory of 16 test cases.